homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -188,8 +188,8 @@ def _disconnect(conn: Connection) -> None:
         try:
             conn.close()
             conn.requests_session.close()
-        except Exception:  # noqa: BLE001
-            _LOGGER.debug("Disconnect error", exc_info=True)
+        except Exception:
+            _LOGGER.exception("Disconnect error")
```

Comment:
I think this will add some unnecessary noise to logs, as there's likely nothing we can generally do if the disconnect fails, and IIRC it is or at least was a common phenomenon with some boxes/firmwares/control flows. I'd be fine with logging the exceptions we know we can do nothing about and that are known to be harmless with .debug and others with .exception (I can't tell which of them are such offhand, and don't have time to figure out/work on this any time soon), but I'm not quite convinced this change would be an improvement as is.

Reply:
But I think if there are exceptions that we did not catch, which do have a reason, we should be clearer to the user. For example if we suddenly get a TimeoutError, either tell the user we cannot connect, or we timed out, both are better than just "Unexpected error".

These excepts are usually just here to make sure the user is guaranteed not to see a "500 - invalid flow specified" (which I think is also something we can fix in the frontend to be more graceful)

Reply:
I agree with Joost